File: scala/src/main/scala/com/ngt/source/KafkaSourceDemo.scala
package com.ngt.source

import org.apache.flink.api.common.serialization.SimpleStringSchema
import org.apache.flink.streaming.api.scala._
import org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer

import java.util.Properties

/**
 * @author ngt
 * @create 2021-01-26 19:15
 */
object KafkaSourceDemo {
  def main(args: Array[String]): Unit = {
    val env: StreamExecutionEnvironment = StreamExecutionEnvironment.getExecutionEnvironment

    val properties: Properties = new Properties()
    properties.setProperty("bootstrap.servers", "hadoop102:9092,hadoop103:9092,hadoop104:9092")
    //指定组ID
    properties.setProperty("group.id", "gwc10")
    //如果没有记录偏移量，第一次从最开始消费
    properties.setProperty("auto.offset.reset", "earliest")
    //kafka的消费者不自动提交偏移量
    //properties.setProperty("enable.auto.commit", "false")

    val lines: DataStream[String] =
      env.addSource(new FlinkKafkaConsumer[String]("wc10", new SimpleStringSchema(), properties))

    lines.print()
    env.execute()
  }
}

File: java/src/main/java/com/ngt/source/KafkaSourceDemo.java
package com.ngt.source;

import org.apache.flink.api.common.serialization.SimpleStringSchema;
import org.apache.flink.streaming.api.datastream.DataStream;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer;

import java.util.Properties;

/**
 * @author ngt
 * @create 2021-01-25 15:15
 */
public class KafkaSourceDemo {
    public static void main(String[] args) throws Exception {
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();

        Properties props = new Properties();
        //指定Kafka的Broker地址
        props.setProperty("bootstrap.servers", "hadoop102:9092,hadoop103:9092,hadoop104:9092");
        //指定组ID
        props.setProperty("group.id", "gwc10");
        //如果没有记录偏移量，第一次从最开始消费
        props.setProperty("auto.offset.reset", "earliest");
        //kafka的消费者是否自动提交偏移量
        //props.setProperty("enable.auto.commit", "false");


        //KafkaSource
        FlinkKafkaConsumer<String> kafkaSource = new FlinkKafkaConsumer<>(
                "wc10",
                new SimpleStringSchema(),
                props);

        //Source
        DataStream<String> lines = env.addSource(kafkaSource);

        //Sink
        lines.print();

        env.execute("KafkaSource");
    }
}
